tensorflow/python/estimator/inputs/pandas_io.py

```diff
@@ -116,6 +116,8 @@ def input_fn():
     features = dict(zip(list(x.columns), features))
     if y is not None:
       target = features.pop(target_column)
+      # undo modification to dictionary
+      del x[target_column]
```

Comment:
Line 79 of this file has x=x.copy(). So, same question: Do you really need this line?